### DISCUSSION TITLE
Frame buffer

### DISCUSSION THREAD

**Discussion by wh1ter0z** [Bodmer/TFT_eSPI#3030] [Q&A - General] [answered]
Hi 

Is there any func to get buffer of tft and share that to web ? 

Target: tft screen share

**Comment by Bodmer** [comment]
If your screen supports reading the CGRAM then you can read out that memory. You can test read compatibility by running this example:

https://github.com/Bodmer/TFT_eSPI/tree/master/examples/Test%20and%20diagnostics/TFT_ReadWrite_Test

Creating and encoding the screen pixels into a suitable image format and handling the web transfer would have to be handled in the sketch. There are PNG and jpeg decoder Arduino libraries available that may be of use.

**Reply by Bodmer** [reply, marked as answer]
Anothe roption if your processor has enough RAM is to draw the screen into a sprite (which is simply a RAM image buffer). Note that the library uses 16 bit 565 RGB encoding for colours.
